### PR TITLE
fixit: Initialize direct controller registry for postsubmit samples tests

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

-Fix postsubmit failures for direct controller
-Deprecate old samples test harness

```
--- PASS: TestAll (0.75s)
    --- PASS: TestAll/linear-log-metric (23.70s)
PASS
{"severity":"info","timestamp":"2024-08-15T21:16:13.140Z","msg":"Stopping and waiting for non leader election runnables"}
{"severity":"info","timestamp":"2024-08-15T21:16:13.141Z","msg":"Stopping and waiting for leader election runnables"}
{"severity":"info","timestamp":"2024-08-15T21:16:13.141Z","msg":"Stopping and waiting for caches"}
{"severity":"info","timestamp":"2024-08-15T21:16:13.141Z","msg":"Stopping and waiting for webhooks"}
{"severity":"info","timestamp":"2024-08-15T21:16:13.141Z","logger":"controller-runtime.webhook","msg":"Shutting down webhook server with timeout of 1 minute"}
{"severity":"info","timestamp":"2024-08-15T21:16:13.141Z","msg":"Wait completed, proceeding to shutdown the manager"}
ok      github.com/GoogleCloudPlatform/k8s-config-connector/config/tests/samples/create 34.478s
```

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: config/tests/samples/create/harness.go [hunk 1]
```diff
@@ -145,6 +151,9 @@ func NewHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptio
 	// creating multiple managers for tests will fail if more than one
 	// manager tries to bind to the same port.
 	kccConfig.ManagerOptions.HealthProbeBindAddress = "0"
+	// Enable the configuration of state-into-spec default value in kccmanager
+	kccConfig.StateIntoSpecDefaultValue = k8s.StateIntoSpecDefaultValueV1Beta1
```
Comment (gemmahou) [inline config/tests/samples/create/harness.go:155]: This was added in a previous [PR](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1152) but got removed later(for unknown reasons). I think we want to add it back.

File: pkg/controller/direct/registry/registry.go [hunk 2]
```diff
@@ -70,7 +70,11 @@ func AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error)
 	}
 	return nil, nil
 }
+
 func Init(ctx context.Context, config *config.ControllerConfig) error {
+	if singleton.registrations == nil {
```
Comment (jasonvigil) [inline pkg/controller/direct/registry/registry.go:75]: Curious, why was this added..?

**Review by jasonvigil — APPROVED** [review]
Will wait for @maqiuyujoyce to comment on state-into-spec issue, and for others to review also.

**Comment by jasonvigil** [comment]
Oh lol, didn't realize approving on github was equivalent to /approve + /lgtm.. only wanted to /approve

File: config/tests/samples/create/harness.go [hunk 3]
```diff
@@ -97,33 +98,38 @@ type Harness struct {
...
+	err := os.Setenv("E2E_KUBE_TARGET", "envtest")
+	if err != nil {
+		t.Errorf("error setting kube target: %v", err)
 	}
-	return NewHarnessWithOptions(ctx, t, opts)
+	return newHarnessWithOptions(ctx, t, opts)
 }
 
-func NewHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptions) *Harness {
+func newHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptions) *Harness {
```
Comment (maqiuyujoyce) [inline config/tests/samples/create/harness.go:132]: It's probably out of scope, but this function is super long, and we should try to refactor / shorten it whenever we have a chance.

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for fixing the samples test for SciFi resources, @gemmahou ! Have you verified TF-based/DCL-based resources still work?
